### DISCUSSION TITLE
Homey in combinatie met DSMR

### DISCUSSION THREAD

**Discussion by Joriempie** [dsmrreader/dsmr-reader#1469] [Questions & answers / How-to's] [answered]
Hallo,

Onlangs overgestapt van een andere fabrikant naar DSMR. De handleiding voor de installatie was super duidelijk.

Nu probeer ik mijn Homey te verbinden met DSMR maar dit gaat niet lekker. Ik heb de handleiding doorgenomen zoals [hier](https://community.athom.com/t/p1-smartmeter-api/6150) stond onder het stukje: "Script for reading the data and push it to the api (DSMR-Reader)". Echter krijg ik nog steeds helaas geen waardes door op mijn Homey.

Excuus als ik hier niet aan het juiste adres ben.

**Comment by dennissiemensma** [comment]
Bedankt voor je vraag. 

Heb je dit script in DSMR-reader gezet?
- https://gist.github.com/steffjenl/31bd083eeb9d0be04375b7695b9f2eaf

En vervolgens ook de plugin in je settings.py of envvar van DSMR-reader gezet?
- `DSMRREADER_PLUGINS=dsmr_plugins.modules.forward_raw_telegram_to_api`

En daarna DSMR-reader **herstart**?

Zo ja, check ook even of je foutmeldingen ziet in de `dsmr_backend` log:
- https://dsmr-reader.readthedocs.io/nl/v4/how-to/troubleshooting/logfiles.html

**Comment by Joriempie** [comment]
Thanks voor de snelle reply!

Het stukje met "var" vind ik zelf erg onduidelijk. Maar dat is een persoonlijke kwestie ben ik bang. Met settings.py heb ik nog niet gewerkt. Zou je kunnen aangeven hoe je dit precies moet doen? Ik heb het Homey script inderdaad onder de juiste directory en naam gezet.

**Comment by dennissiemensma** [comment]
Over het inschakelen van die plugin kun je hier meer informatie vinden:
- https://dsmr-reader.readthedocs.io/nl/v4/reference/plugins.html#configuration

Als je het script `forward_raw_telegram_to_api.py` hebt genoemd, dan wordt de instelling dus 

```
DSMRREADER_PLUGINS=dsmr_plugins.modules.forward_raw_telegram_to_api
```

**Comment by Joriempie** [comment, marked as answer]
> Over het inschakelen van die plugin kun je hier meer informatie vinden:
> 
> * https://dsmr-reader.readthedocs.io/nl/v4/reference/plugins.html#configuration
> 
> Als je het script `forward_raw_telegram_to_api.py` hebt genoemd, dan wordt de instelling dus
> 
> ```
> DSMRREADER_PLUGINS=dsmr_plugins.modules.forward_raw_telegram_to_api
> ```

Check. Ik weet al wat mijn probleem is. Ik dacht dat ik een nieuw bestand moest maken met bijvoorbeeld:
`forward_raw_telegram_to_api.env`

Maar dat blijkt niet zo te zijn. Want er staat binnen de hoofd directory: "home/dsmr/dsmr-reader" een `.env` file echter zag ik deze niet staan omdat ik `ls` heb gebruikt in plaats van `ls -la`.

Ik kwam er achter doordat ik debugging aan wou zetten via deze link: https://dsmr-reader.readthedocs.io/en/latest/how-to/troubleshooting/enabling-debug-logging.html echter zag ik:
`### Logging level.
###DSMRREADER_LOGLEVEL=DEBUG` niet staan waarbij een lampje bij mij ging branden. Doormiddel van het commando: `find . -type f -name "*.env"` kwam ik er achter waar het bestandje vond.

**Comment by dennissiemensma** [comment]
@Joriempie fijn om te horen! Wellicht dat het laden van plugins in een toekomstige versie wat makkelijker wordt, maar voor nu is het even puzzelen soms.
